### PR TITLE
Dev/jupyterlab 3.0

### PR DESCRIPTION
**Pull request by joequant**

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
Thanks!

**Comment by blink1073** [comment]
I'll merge an release this on Monday

**Comment by bollwyvl** [comment]
This is great: maybe we hoist `pdfjs-dist` to a `sharedPackages`? Every time i see one of these "big fella" dependencies by a first-party extension, i want to be able to reuse without shipping it again... though really, pdf is likely important enough to live in core at some point, so it's more predictable/computable...

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "lib/*.d.ts",
     "lib/*.js",
     "schema/*.json",
-    "style/*.*"
+    "style/*.*",
+    "style/index.js"
```
Comment (bollwyvl) [inline package.json:28]: this is probably covered by the `*.*` above it...

File: package.json [hunk 2]
```diff
@@ -45,46 +46,56 @@
...
-    "@jupyterlab/services": "^5.0.0",
-    "@jupyterlab/settingregistry": "^2.0.0",
-    "@jupyterlab/statedb": "^2.0.0",
-    "@jupyterlab/ui-components": "^2.2.3",
-    "@lumino/coreutils": "^1.3.1",
-    "@lumino/disposable": "^1.2.0",
-    "@lumino/domutils": "^1.1.3",
-    "@lumino/messaging": "^1.2.3",
-    "@lumino/widgets": "^1.8.0",
+    "@jupyterlab/application": "^3.0.0",
```
Comment (bollwyvl) [inline package.json:49]: if there were really no code changes (which is good), could these all be `^2.0.0 || ^3.0.0`?

File: package.json [hunk 3]
```diff
@@ -45,46 +46,56 @@
...
+    "@jupyterlab/services": "^6.0.0",
+    "@jupyterlab/settingregistry": "^3.0.0",
+    "@jupyterlab/statedb": "^3.0.0",
+    "@jupyterlab/ui-components": "^3.0.0",
+    "@lumino/coreutils": "^1.5.3",
+    "@lumino/disposable": "^1.4.3",
+    "@lumino/domutils": "^1.2.3",
+    "@lumino/messaging": "^1.4.3",
+    "@lumino/widgets": "^1.16.1",
     "pdfjs-dist": "2.0.943",
```
Comment (bollwyvl) [inline package.json:68]: Mmmm... if we bump this to the latest (well, any release after [2.6.347](https://github.com/mozilla/pdf.js/releases/tag/v2.6.347)), we'd get typings...

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/how-to-set-up-development-of-jupyterlab-latex-extension/9294/4

**Comment by aquirdTurtle** [comment]
Is this going to be merged soon, or is it waiting for something?

**Comment by joequant** [comment]
Hi.  Anything needed to merge the PR? for jupyterlab 3.0?

**Comment by blink1073** [comment]
> Hi. Anything needed to merge the PR? for jupyterlab 3.0?

@bollwyvl had given some suggestions above, are you able to make those changes @joequant?

**Comment by aquirdTurtle** [comment]
I hate to ask, but are these minor changes really worth waiting for, or can't someone from the jupyterlab team make the changes? This extension is already ~6 months late coming to jupyterlab 3...

**Comment by blink1073** [comment]
I'm going to cut a release with this progress, thanks @joequant!

**Comment by blink1073** [comment]
@mpacer, @ian-r-rose, can one of you please add `blink1073` to https://pypi.org/project/jupyterlab-latex/?

**Comment by blink1073** [comment]
npm package is [up](https://www.npmjs.com/package/@jupyterlab/latex/v/3.0.0), but I can't yet publish to PYPI.

**Comment by blink1073** [comment]
Finished, thanks @mpacer!

https://pypi.org/project/jupyterlab-latex/3.0.0/

**Pull request merged**